apps/server/src/modules/account/services/account.service.ts

```diff
@@ -34,119 +48,267 @@ export class AccountService extends AbstractAccountService {
 		}
 	}
 
-	async findById(id: string): Promise<AccountDto> {
+	public async updateMyAccount(user: User, account: Account, updateData: UpdateMyAccount) {
+		if (account.systemId) {
+			throw new ForbiddenOperationError('External account details can not be changed.');
+		}
+
+		if (!updateData.passwordOld || !(await this.validatePassword(account, updateData.passwordOld))) {
+			throw new AuthorizationError('Your old password is not correct.');
+		}
+
+		let updateUser = false;
+		let updateAccount = false;
+		if (updateData.passwordNew) {
+			account.password = updateData.passwordNew;
+			updateAccount = true;
+		} else {
+			account.password = undefined;
+		}
+
+		if (updateData.email && user.email !== updateData.email) {
+			const newMail = updateData.email.toLowerCase();
+			await this.checkUniqueEmail(account, user, newMail);
+			user.email = newMail;
+			account.username = newMail;
+			updateUser = true;
+			updateAccount = true;
+		}
+
+		if (updateData.firstName && user.firstName !== updateData.firstName) {
+			user.firstName = updateData.firstName;
+			updateUser = true;
+		}
+
+		if (updateData.lastName && user.lastName !== updateData.lastName) {
+			user.lastName = updateData.lastName;
+			updateUser = true;
+		}
+
+		if (updateUser) {
+			try {
+				await this.userRepo.save(user);
```

Comment:
The userRepo is not avaible for you. Only the service, do not use it directly.
I think the service / repo of the user throw errors for you in correct way. Do not really have a need to re-throw it. Re-throw without passing the root error is invalid.

const promises = [];

if (updateUser) promise.push(xx)
if (updateAccount) promise.push(yy)

const promiseStatus = await Promise.settled(promise);
// handle status of rejection